packages/@vue/cli-service/lib/config/base.js

```diff
@@ -18,6 +18,9 @@ module.exports = (api, options) => {
         .rule('esm')
           .test(/\.m?jsx?$/)
           .resolve.set('fullySpecified', false)
+
+      webpackConfig
+        .cache({ type: 'filesystem' })
```

Comment:
Don't do this. It's not safe. I'm pretty sure it'll fail at least one of these test cases:
https://github.com/sodatea/vue-cli/blob/e3cfcf57a30bb58ccc7895fe1d3242fcef797501/packages/%40vue/cli-service/__tests__/cache.spec.js#L32-L61

Reply:
Thanks for the feedback! I appreciate it.

It's not explicitly stated in the documentation, but the spec for caching in webpack 5 states: 
> Each build with a different webpack configuration should store a unique copy of its cache versus another webpack configuration. E.g. A development build and a production build must have distinct caches due to them having different options set.

The spec for it is here: https://github.com/webpack/webpack/issues/6527

I'm not sure how the `--mode` option works in vue-cli, but it does look like those specific tests did not fail for this PR. You can see the reports for those here: https://app.circleci.com/pipelines/github/vuejs/vue-cli/1124/workflows/10aeafe8-27a2-486a-9609-2111481209be/jobs/27240

The only two tests that failed do not seem to be related to the changes I made.

Reply:
The link points to another PR that I'm working on.
The tests are not written yet, therefore it's not in the pipeline.

Reply:
> The only two tests that failed do not seem to be related to the changes I made.

I'm afraid they're related. Looks like the build process for modern mode reused the cache from the legacy mode.